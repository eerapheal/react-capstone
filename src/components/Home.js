import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import {
  faCircleArrowRight,
  faSearch,
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { getCountries } from '../redux/Details';

const Home = () => {
  const dispatch = useDispatch();
  const { countries, status } = useSelector((state) => state.datas);
  const [search, setSearch] = useState('');

  useEffect(() => {
    if (status === null) {
      dispatch(getCountries());
    }
  });

  return (
    <div>
      {status === 'pending' ? (
        <div className="pending">
          <h3> Loading...</h3>
        </div>
      ) : (
        <>
          <div className="search">
            <div className="search_element">
              <h1 className="live_preview">Live preview</h1>
              <input
                type="text"
                value={search}
                onChange={(e) => setSearch(e.target.value)}
                placeholder="search"
                className="search_input"
              />
              <FontAwesomeIcon icon={faSearch} className="search_icon" />
            </div>
          </div>
          <div className="display_flex">
            {countries
              .filter(
                (searchCountry) => searchCountry.continent
                  .toLowerCase()
                  .includes(search.toLowerCase())
                || searchCountry.continent
                  .toLowerCase()
                  .includes(search.toLowerCase()),
              ).map((item) => (
                <div key={uuidv4()} className="country_display">
                  <NavLink state={item} to="/details">
                    <FontAwesomeIcon
                      icon={faCircleArrowRight}
                      className="arrow-right"
                    />
                  </NavLink>
                  <div className="update_countries">
                    <h3>{item.continent}</h3>
                    <p>{item.updated}</p>
                  </div>
                </div>
              ))}
          </div>
        </>
      )}
    </div>
  );
};

export default Home;
