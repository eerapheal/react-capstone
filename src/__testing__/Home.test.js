import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../components/Home';

describe('snapshot testing', () => {
  test('snapshot for Home', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
