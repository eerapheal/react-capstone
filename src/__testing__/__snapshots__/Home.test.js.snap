// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Home 1`] = `
<div>
  <div
    className="search"
  >
    <div
      className="search_element"
    >
      <h1
        className="live_preview"
      >
        Live preview
      </h1>
      <input
        className="search_input"
        onChange={[Function]}
        placeholder="search"
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-magnifying-glass search_icon"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="display_flex"
  />
</div>
`;
